lib/Models/Workbench.ts

```diff
@@ -167,14 +166,15 @@ export default class Workbench {
    *
    * @param item The item to add to or remove from the workbench.
    */
-  public async add(item: BaseModel | BaseModel[]): Promise<void> {
+  public async add(
+    item: BaseModel | BaseModel[],
+    index: number = 0
+  ): Promise<void> {
     if (Array.isArray(item)) {
-      await Promise.all(item.reverse().map(i => this.add(i)));
+      await Promise.all(item.reverse().map(i => this.add(i, index)));
```

Comment:
wait why do we need to reverse the item array?

Reply:
If I recall correctly, it is done so the item gets placed in the correct position in the workbench. 
When an array is supplied to the `add` function items should be added in the same order so the first array item gets at the top of the workbench (or desired index), without reverse that order would be messed up